scripts/restream.py

```diff
@@ -0,0 +1,108 @@
+# invoke from root as python -m scripts.restream
+import sys
+from subprocess import Popen, PIPE
+import time
+import re
+
+with open("config.stream.template.json", "r") as content_file:
+    config_base_file = content_file.read()
+
+
+RESOLUTIONS = {
+    "1080": [1920, 1080],
+    "720": [1280, 720],
+    "480": [852, 480],
+    "360": [640, 360],
+}
+
+CAP_RATIOS = [1233 / 1920, 1]  # based on the stencil dimensions for 1080p
+
+PLAYER_SETTINGS = [
+    [8081, 4001],
+    [8082, 4002],
+]
+
+
+def setupPlayer(twitch_name, player_num):
+    local_stream_port, ocr_dest_port = PLAYER_SETTINGS[player_num - 1]
+
+    twitch_url = "twitch.tv/{}".format(twitch_name)
+    local_url = "http://localhost:{}".format(local_stream_port)
+
+    fps = -1
+
+    resolutions = [
+        "720p",
+        "480p",
+        "360p",
+        "720p60",
+        "480p60",
+        "1080p",
+        "1080p60",
+    ]
+
+    # run local restreamer over http
+    Popen(
+        [
+            "streamlink",
+            twitch_url,
+            ",".join(resolutions),
+            "--player",
+            "vlc --intf dummy --sout '#standard{access=http,mux=mkv,dst=localhost:"
+            + str(local_stream_port)
+            + "}'",
+        ]
+    )
+
+    time.sleep(10)
+
+    p = Popen(["ffmpeg", "-i", local_url], stdout=PIPE, stderr=PIPE)
+
+    stdout, stderr = p.communicate()
+
+    fps_re = re.compile(", (?P<fps>[0-9.]+) tbr,")
+
+    m = fps_re.search(str(stderr))
+
+    try:
+        fps = int(m.group("fps"))
+    except Exception:
+        fps = float(m.group("fps"))
+
+    print("Found fps", fps)
+
+    size_re = re.compile(", \\d+x(?P<size_key>(360|480|720|1080))(,| \\[SAR)")
+
+    m = size_re.search(str(stderr))
+
+    width, height = RESOLUTIONS[m.group("size_key")]
+
+    print("Found size", width, height)
+
+    # Update player config file
+    player_config = (
+        config_base_file.replace("USER_NAME", twitch_name)
```

Comment:
This hack is both awesome and eye sore inducing at the same time 😛 JSON with variable is used extensively in Kubernetes, as a "language" called Jsonnet. A Python sample can be found at: https://jsonnet.org/ref/bindings.html#python_api

If you are okay with it, I can issue a subsequent PR later to port your JSON file to `.jsonnet` and update the chunk of code here 😄

Reply:
agree that string-replace is hackerman. Why not just init player_config, then do
```
config["user.user_name"] = twitch_name
config["user.twitch_name"] = twitch_name
config["perf.scan_rate"] = fps
```
etc?

Reply:
Also i've added a commandline hook `--defaultconfig` so you can pass in `config.stream.template.json` into it. 
That might be helpful (or might not be :( )

I could see that chucking `--defaultconfig config.stream.template.json` and `--config config.competition.p{}.json` together, or adding them as args to Config's constructor, and then modifying then calling Config.SaveAs()

Reply:
Haha, I did consider doing proper config edit (json deserialize, edit local object, json reserialize), but I was looking at the smallest fastest way to get to where I wanted 😅 

Zero objection to using jsonnet instead! :)

Reply:
Cleaned up with using the `Config` class 👍